### PR TITLE
Ticket #8416 - Ticket #9185 - Ticket #7437 - MailReport Package

### PR DESCRIPTION
**Pull request by joshuasign**
Add minute management - Ticket #8416
Correction to be able to send over multiples addresses - Ticket #9185
Change TLS autostart to be always true, it should be the better default value in any case, as explained in the phpmailer class. - Ticket #7437

### CODE REVIEW

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 1]
```diff
@@ -62,7 +62,7 @@ function set_mail_report_cron_jobs($a_mailreports) {
 	// Add new cron jobs
 	foreach($a_mailreports as $id => $report) {
 		$cron_item = array();
-		$cron_item['minute'] = 0;
+		$cron_item['minute'] = isset($report['minuteofday']) ? $report['minuteofday'] : 0;;
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: You have an extra `;` on this line.

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 2]
```diff
@@ -80,6 +80,9 @@ function mail_report_send($headertext, $cmdtext, $logtext, $attachments) {
 
 	if (empty($config['notifications']['smtp']['ipaddress']))
 		return;
+	
+	if (empty($config['notifications']['smtp']['notifyemailaddress']))
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: Please use ["K&R, BSD KNF variant style" formatting](https://www.netgate.com/docs/pfsense/development/developer-style-guide.html#indent-style). I know you were just matching what was already there, but it's better to do it right.

You can fix the other existing code if you like, in a separate commit.

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 3]
```diff
@@ -105,8 +108,12 @@ function mail_report_send($headertext, $cmdtext, $logtext, $attachments) {
 	$mail->IsHTML(true);
 	$mail->AddReplyTo($config['notifications']['smtp']['fromaddress'], "Firewall Email Report");
 	$mail->SetFrom($config['notifications']['smtp']['fromaddress'], "Firewall Email Report");
-	$address = $config['notifications']['smtp']['notifyemailaddress'];
-	$mail->AddAddress($address, "Report Recipient");
+	$addresses = explode(",", $config['notifications']['smtp']['notifyemailaddress']);
+	foreach ($addresses as $address){
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: Needs a space between `)` and `{`,

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 4]
```diff
@@ -105,8 +108,12 @@ function mail_report_send($headertext, $cmdtext, $logtext, $attachments) {
...
+		$recipient=explode("@", $address);
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: Needs spaces around `=`.

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 5]
```diff
@@ -105,8 +108,12 @@ function mail_report_send($headertext, $cmdtext, $logtext, $attachments) {
...
+		$recipient=str_replace(array("_",".")," ",$recipient[0]);
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: Needs spaces around `=` and after `,` between parameters.

File: mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc [hunk 6]
```diff
@@ -105,8 +108,12 @@ function mail_report_send($headertext, $cmdtext, $logtext, $attachments) {
...
+		$mail->AddAddress($address, ucwords(strtolower($recipient));
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/etc/inc/mail_reports.inc]: You are missing a closing `)` before `;`

File: mail/pfSense-pkg-mailreport/files/usr/local/www/status_mail_report_edit.php [hunk 7]
```diff
@@ -149,7 +149,8 @@
 
 	// Default to midnight if unset/invalid.
 	$pconfig['timeofday'] = isset($pconfig['timeofday']) ? $pconfig['timeofday'] : 0;
-	$friendlytime = sprintf("%02d:00", $pconfig['timeofday']);
+	$pconfig['minuteofday'] = isset($pconfig['minuteofday']) ? $pconfig['minuteofday'] : 0;
+	$friendlytime = sprintf("%02d:%02d", $pconfig['timeofday'],$pconfig['minuteofday']);
```
Comment (jim-p) [inline mail/pfSense-pkg-mailreport/files/usr/local/www/status_mail_report_edit.php]: Needs a space after `,` before `$pconfig['minuteofday']`.

**Comment by joshuasign** [comment]
I'am looking for adding the makefile to this request without creating a new one.
I'll add it as soon as possible.

**Pull request merged**